Fix InfoLayoutHelper not using first video stream, update details when selecting version

StubbyB opened the pull request:
On Android TV some tags are shown such as 'R', '4K' 'HEVC', etc.
Titles which have external subtitles, are missing the resolution and video codec tags (e.g. '1080' and 'H264').

The source of the issue is `org.jellyfin.androidtv.util.InfoLayoutHelper.addRatingAndRes` getting all media streams, and assuming that the video stream will be the first one, when in fact external subtitles are first (also verifiable by looking at the debug info at the bottom).

**Changes**
Instead of assuming the video is the first stream, take the first stream tagged as `MediaStreamType.VIDEO`.

**Issues**
#2811

nielsvanvelzen reviewed (changes requested):
We already have a `StreamHelper.getFirstAudioStream` function so it would probably make sense to add a `StreamHelper.getFirstVideoStream` or maybe even merge them to a `StreamHelper.getFirstStreamOfType()`.

StubbyB commented:
> We already have a `StreamHelper.getFirstAudioStream` function so it would probably make sense to add a `StreamHelper.getFirstVideoStream` or maybe even merge them to a `StreamHelper.getFirstStreamOfType()`.

Moved logic to `StreamHelper`.
Also, since the order of the video versions in the selection menu is the order of the media sources, I added the functionality of settings the tags given a selected media source index, with 0 being the default (preserving existing behavior).
Had to add function signatures accordingly.

StubbyB commented:
made a few cosmetic changes, done modifying until reviewed :) @nielsvanvelzen

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/MyDetailsOverviewRow.kt:
```diff
@@ -14,6 +14,7 @@ class MyDetailsOverviewRow @JvmOverloads constructor(
 	var infoItem1: InfoItem? = null,
 	var infoItem2: InfoItem? = null,
 	var infoItem3: InfoItem? = null,
+	var selectedMediaSourceIndex: Int = 0
```
```suggestion
	var selectedMediaSourceIndex: Int = 0,
```

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/util/InfoLayoutHelper.java:
```diff
@@ -45,17 +45,20 @@ public static void addInfoRow(Context context, BaseRowItem item, LinearLayout la
     }
-
-    public static void addInfoRow(Context context, BaseItemDto item, LinearLayout layout, boolean includeRuntime, boolean includeEndTime) {
+    public static void addInfoRow(Context context, BaseItemDto item, int mediaSourceIndex, LinearLayout layout, boolean includeRuntime, boolean includeEndTime) {
```
(missing the empty line before the function)
```suggestion

    public static void addInfoRow(Context context, BaseItemDto item, int mediaSourceIndex, LinearLayout layout, boolean includeRuntime, boolean includeEndTime) {
```

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/util/InfoLayoutHelper.java:
```diff
@@ -325,14 +328,17 @@ private static void addDate(@NonNull Context context, BaseItemDto item, LinearLa
+
+        MediaStream videoStream = StreamHelper.getFirstVideoStream(item, mediaSourceIndex);
+
+        if(videoStream != null && videoStream.getWidth() != null && videoStream.getHeight() != null) {
```
```suggestion
        if (videoStream != null && videoStream.getWidth() != null && videoStream.getHeight() != null) {
```

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:
```diff
@@ -26,9 +26,33 @@ public static List<MediaStream> getAudioStreams(MediaSourceInfo mediaSource) {
+        return getFirstAudioStream(item, 0);
+    }
+
+    public static MediaStream getFirstAudioStream(BaseItemDto item, int mediaSourceIndex) {
```
```suggestion
    public static MediaStream getAudioStream(BaseItemDto item, int mediaSourceIndex) {
```
It's not the first if we explicitly tell it the index.

StubbyB replied:
It is the first matching `MediaStream` for the given `MediaSource`.

The idea was to allow the existing functionality of retrieving the first overall media stream, and also an overload allowing to choose a specific source.

Also `getFirstAudioStream` is an existing public util which I was reluctant to change.

Would you rather it be a separate function signature named something like:
`getFirstStreamOfSource(BaseItemDto item, int mediaSourceIndex)`
`getFirstStreamOfSource(BaseItemDto item, int mediaSourceIndex, MediaStreamType streamType)`?

StubbyB replied:
This applies to all other comments ofc

nielsvanvelzen replied:
Aah you're right, I missed that part. This naming is fine 👍.

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:
```diff
@@ -26,9 +26,33 @@ public static List<MediaStream> getAudioStreams(MediaSourceInfo mediaSource) {
+        return getFirstVideoStream(item, 0);
+    }
+
+    public static MediaStream getFirstVideoStream(BaseItemDto item, int mediaSourceIndex) {
```
```suggestion
    public static MediaStream getVideoStream(BaseItemDto item, int mediaSourceIndex) {
```

nielsvanvelzen commented on app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:
```diff
@@ -26,9 +26,33 @@ public static List<MediaStream> getAudioStreams(MediaSourceInfo mediaSource) {
+        return getFirstStreamOfType(item, streamType, 0);
+    }
+
+    public static MediaStream getFirstStreamOfType(BaseItemDto item, MediaStreamType streamType, int mediaSourceIndex) {
```
```suggestion
    public static MediaStream getStreamOfType(BaseItemDto item, MediaStreamType streamType, int mediaSourceIndex) {
```

nielsvanvelzen commented:
Thank you!
